wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ClusterAwareReaderFailoverHandler.java

```diff
@@ -370,6 +363,20 @@ private ReaderFailoverResult getResultFromNextTaskBatch(
     return new ReaderFailoverResult(null, null, false);
   }
 
+  protected FullServicesContainer getNewServicesContainer() throws SQLException {
+    return ServiceUtility.getInstance().createServiceContainer(
```

Comment:
Just a suggestion. I've seen such piece of code at least twice. I think it makes sense to create a new method:
`ServiceUtility.createNewFromSource(FullServicesContainer source)`